main.tf

```
@@ -205,7 +205,7 @@ module "hostname" {
   version = "0.12.2"
 
   enabled  = module.this.enabled && length(var.zone_id) > 0
-  dns_name = "${module.this.name}-broker-${count.index + 1}"
+  dns_name = "${module.this.id}-broker-${count.index + 1}"
```

```suggestion
  dns_name = "${var.broker_full_id_record_enabled ? module.this.id : module.this.name}-broker-${count.index + 1}"
```

Variable block (after `var.broker_volume_size`):

```hcl
variable "broker_full_id_record_enabled" {
  type        = bool
  default     = false
  description = <<-EOT
  Whether to use the module's ID (`module.this.id`) in the broker record. If set to `false`, `var.name` will be used instead.

  It is recommended that this is set to `true` when the Hosted Zone name does not provide any information on where the broker
  is deployed. For example, if the HZ is `example.local`, then set this to `true` in order to produce a broker record such
  as `eg-ue1-dev-msk-broker-1.example.local` instead of `msk-broker-1.example.local`.

  However, if the HZ name provides more information, such `ue1.dev.example.com`, then set this to `false` in order to produce
  a broker record such as `msk-broker-1.ue1.dev.example.com` instead of `eg-ue1-dev-msk-broker-1.ue1.dev.example.com`.
  EOT
}
```